sdk/keyvault/keyvault-keys/migration-guide.md

```diff
@@ -0,0 +1,251 @@
+
+
+```js
+var KeyVault = require('azure-keyvault');
+var msRestAzure = require('ms-rest-azure');
+
+const clientId = "client id";
+const secret = "client secret";
+const domain = "tenant id";
+const vaultUrl = `https://my-vault.vault.azure.net/`;
+
+async function main() {
+  const credentials = await msRestAzure.loginWithServicePrincipalSecret(
+    clientId,
+    secret,
+    domain
+  );
+  const client = new KeyVault.KeyVaultClient(credentials);
+  const keyVaultKey = await client.getKey(vaultUrl, "MyKey", "");
+  console.log(keyVaultKey);
+}
+
+main().catch((err) => console.error(err));
+```
+
+Now in `@azure/keyvault-keys` you can create a `KeyClient` using any credential from [`@azure/identity`][identity-readme].
+
+You can install them by simply running the following command at the root of your project:
+
+```
+npm install --save @azure/identity @azure/keyvault-keys
+```
+
+Below is a simple example using both `@azure/keyvault-keys` and [`DefaultAzureCredential`][identity-readme-DAC]:
+
+```ts
+// The default credential first checks environment variables for configuration as described above.
+// If environment configuration is incomplete, it will try managed identity.
+const { KeyClient } = require("@azure/keyvault-keys");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+// Azure SDK clients accept the credential as a parameter
+const credential = new DefaultAzureCredential();
+const vaultUrl = "https://my-vault.vault.azure.net/";
+const client = new KeyClient(vaultUrl, credential);
+
+async function main(): Promise<void> {
+  const keyVaultKey = await client.getKey("MyKey");
+  console.log(keyVaultKey);
+}
+
+main().catch((err) => console.error(err));
+```
+
+The new client also provides a `CryptographyClient` to perform cryptographic operations (encrypt/decrypt, wrap/unwrap, sign/verify) using a particular key. Below is an example.
+
+```ts
+const { KeyClient, CryptographyClient } = require("@azure/keyvault-keys");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+const credential = new DefaultAzureCredential();
+const vaultUrl = "https://my-vault.vault.azure.net/";
+const client = new KeyClient(vaultUrl, credential);
+
+async function main(): Promise<void> {
+  const keyVaultKey = await client.getKey("MyKey");
+  const cryptographyClient = new CryptographyClient(keyVaultKey.id!, credential);
+  const signed = await cryptographyClient.signData("RS256", "some data");
+  console.log(signed);
+}
+
+main().catch((err) => console.error(err));
+```
+
+### Create a key
+
+In `azure-keyvault` you could create a key by using `KeyVaultClient`'s `createKey` method, which required a vault endpoint, key name, and key type. This method returned a `KeyBundle` containing the key.
+
+```js
+// create an RSA key
+let keyBundle = await client.createKey(vaultUrl, "myRSAKey", "RSA");
+console.log(keyBundle);
+
+// create an elliptic curve key
```

Comment:
These comments are nice and useful, please add them to the Track 2 code snippet as well